### ISSUE TITLE
Maui -Android certificate pinning using network-security-config ignore incorrect pin

### ISSUE THREAD

**Issue by DmitryNikiforovCoherent** [dotnet/runtime#107695]
### Description

Added ssl pinning using network-security-config

For test change one letter of base 64 pin, but SslClientAuthenticationOptions.RemoteCertificateValidationCallback still pass SslPolicyErrors.None

### Steps to Reproduce

-  Added network_security_config.xml as AndroidResource `<network-security-config>
    <domain-config>
        <domain includeSubdomains="true">{host}</domain>
        <pin-set expiration="2025-03-21">
            <pin digest="SHA-256">pin</pin>
        </pin-set>
    </domain-config>
</network-security-config>` 

- Added SocketsHttpHandler with SslClientAuthenticationOptions.RemoteCertificateValidationCallback
- I expect, that if I change base64 pin to incorrect, RemoteCertificateValidationCallback will be called with RemoteCertificateNameMismatch, as for IOS, but receive SslPolicyErrors.None

### Version with bug

8.0.70 SR7

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Any android version

**Comment by DmitryNikiforovCoherent** [comment]
> There is a bit of confusion on this thread, so I'm not sure I follow what the problem is. Is there a sample that displays a problem?
> 
> One note:
> 
> * Using `AndroidMessageHandler` (default), probably no `RemoteCertificateValidationCallback` would ever be called. It would be using the native Android http stack in that case.
> * Using `SocketsHttpHandler` (you'd have to set `UseNativeHttpHandler=false` in your `.csproj`), `RemoteCertificateValidationCallback` should get called
> 
> @simonrozsival might be able to help, if we think there is a runtime issue here.

In demo I used HttpClient Implementation: Managed(HttpClientHandler)
<img width="498" alt="image" src="https://github.com/user-attachments/assets/e9dcc27d-5603-4998-bcc8-d4a689220113">

Both SslClientAuthenticationOptions.RemoteCertificateValidationCallback and HttpClientHandler.ServerCertificateCustomValidationCallback called, if make a request via them

**Comment by simonrozsival** [comment]
@DmitryNikiforovCoherent thanks for reporting this issue. I looked into it today and I think I have a rough idea where the problems in both the `HttpClientHandler`(internally implemented via `AndroidMessageHandler`) and `SocketsHttpHandler` originate.

Currently the only workaround is to manually check the certificate hash in the `ServerCertificateValidationCallback`/`RemoteCertificateValidationCallback` instead of using `network_security_config.xml`. I would point out that Google doesn't recommend using certificate pinning in Android apps (see https://developer.android.com/privacy-and-security/security-ssl#Pinning).

**Comment by SidlerD123** [comment]
@simonrozsival We have switched from HttpClientHandler to SocketsHttpHandler in our App. After this change, we noticed that our network_security_config.xml does not seem to be respected anymore. From my (limited) understanding, the linked pull request would fix this. Is this correct?

**Comment by simonrozsival** [comment]
@SidlerD123 yes, this PR has been merged and it will be part of the .NET 11 release.
